### PR TITLE
Add OpenMpi to CBL-Mariner

### PR DESCRIPTION
**Pull request by rikenm1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Adding OpenMPI to CBL-Mariner

###### Change Log  

- Change: Add OpenMpi to CBL-Mariner
- Change: move hwloc, pmix, opensm, orangefs, rpm-mpi-hooks, infinipath-sm, libpsm2, environment-modules, and munge to Core.
- Change: update hwloc, pmix, rpm-mpi-hooks, and environment-modules

###### Does this affect the toolchain?  

**YES/NO**
NO

###### Test Methodology

- Pipeline build id: [304126 [AMD]](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=304126&view=logs&s=444ee1b5-8b3f-5598-a629-e88326f0c2f9)
- Pipeline build id: [304109 [ARM]](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=304109&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774)

### CODE REVIEW

File: SPECS/environment-modules/environment-modules.signatures.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+ "Signatures": {
+  "modules-5.2.0.tar.bz2": "65f6537e770c66036c8adcad57c51a389f805d19fe942c963bd9f29337832b6e"
```
Comment (PawelWMS) [inline SPECS/environment-modules/environment-modules.signatures.json:3]: As part of this change, let's also re-name the tarball, so it's clearer what they are for at the same reducing a risk of conflict. I'd suggest the good old `%{name}-%{version}.tar.bz2`.;)

File: SPECS/environment-modules/environment-modules.spec [hunk 2]
```diff
@@ -87,108 +90,91 @@ touch %{buildroot}%{_sysconfdir}/profile.d/modules.{csh,sh}
...
-%{_mandir}/man1/module-compat.1.gz
-%{_mandir}/man4/modulefile-compat.4.gz
-
+%dir %{_datadir}/Modules/nagelfar
+%{_datadir}/Modules/nagelfar/*
 
 %changelog
+* Fri Feb 03 2023 Riken Maharjan <rmaharjan@microsoft.com> - 5.2.0-1
+- Move from extended to Core.
+- Update to 5.2.0 (from Fedora 38(license: MIT)).
```
Comment (PawelWMS) [inline SPECS/environment-modules/environment-modules.spec:174]: Thank you for remembering about the source attribution!

File: SPECS/munge/munge.spec [hunk 3]
```diff
@@ -61,12 +57,11 @@ echo "d /run/munge 0755 munge munge -" > src/etc/munge.tmpfiles.conf.in
 # Get rid of some rpaths for /usr/sbin
 sed -i 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' libtool
 sed -i 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' libtool
-make %{?_smp_mflags} 
+make %{?_smp_mflags}
```
Comment (PawelWMS) [inline SPECS/munge/munge.spec:60]: Since we're here:
```suggestion
%make_build
```

File: SPECS/munge/munge.spec [hunk 4]
```diff
@@ -61,12 +57,11 @@ echo "d /run/munge 0755 munge munge -" > src/etc/munge.tmpfiles.conf.in
...
 
 
 %install
 
-rm -rf %{buildroot}
 make install DESTDIR=%{buildroot}
```
Comment (PawelWMS) [inline SPECS/munge/munge.spec:65]: Similar suggestion: let's update this line to `%make_install`.

File: SPECS/munge/munge.spec [hunk 5]
```diff
@@ -75,24 +70,24 @@ install -p -D -m 644 munge.logrotate %{buildroot}/%{_sysconfdir}/logrotate.d/mun
 
 # rm unneeded files.
 rm %{buildroot}/%{_sysconfdir}/sysconfig/munge
-# 
+#
```
Comment (PawelWMS) [inline SPECS/munge/munge.spec:73]:
```suggestion
```

File: SPECS/openmpi/openmpi.spec [hunk 6]
```diff
@@ -0,0 +1,1139 @@
...
+Source3:        openmpi.pth.py3
+Source4:        macros.openmpi
+BuildRequires:  gcc-c++
+BuildRequires:  gcc-gfortran
+# Doesn't compile:
+# vt_dyn.cc:958:28: error: 'class BPatch_basicBlockLoop' has no member named 'getLoopHead'
+#                      loop->getLoopHead()->getStartAddress(), loop_stmts );
+#BuildRequires:  dyninst-devel
+BuildRequires:  hwloc-devel
+# Old libevent causes issues
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:50]: Let's remove comments, which doesn't mean anything in the context of Mariner. This one looks like an old comment, which may make sense for Fedora devs, but not in our case.
```suggestion
```

File: SPECS/openmpi/openmpi.spec [hunk 7]
```diff
@@ -0,0 +1,1139 @@
...
+Source0:        https://www.open-mpi.org/software/ompi/v4.1/downloads/openmpi-%{version}.tar.bz2
+Source1:        openmpi.module.in
+Source3:        openmpi.pth.py3
+Source4:        macros.openmpi
+BuildRequires:  gcc-c++
+BuildRequires:  gcc-gfortran
+# Doesn't compile:
+# vt_dyn.cc:958:28: error: 'class BPatch_basicBlockLoop' has no member named 'getLoopHead'
+#                      loop->getLoopHead()->getStartAddress(), loop_stmts );
+#BuildRequires:  dyninst-devel
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:48]: Since we're starting without `dyninst-devel`, we might as well remove the comment to clean-up a bit.
```suggestion
```

File: SPECS/openmpi/openmpi.spec [hunk 8]
```diff
@@ -0,0 +1,1139 @@
...
+BuildRequires:  libtool
+BuildRequires:  perl(Data::Dumper)
+BuildRequires:  perl(File::Find)
+%endif
+%if %{with rdma}
+BuildRequires:  opensm-devel
+BuildRequires:  rdma-core-devel
+%endif
+# So configure can find lstopo
+# BuildRequires:  hwloc-gui
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:77]:
```suggestion
```

File: SPECS/openmpi/openmpi.spec [hunk 9]
```diff
@@ -0,0 +1,1139 @@
...
+Requires:       %{name}-java = %{version}-%{release}
+Requires:       java-devel
+
+%description java-devel
+Contains development wrapper for compiling Java with openmpi.
+%endif
+
+# We set this to for convenience, since this is the unique dir we use for this
+# particular package, version, compiler
+%global namearch openmpi-%{_arch}%{?_cc_name_suffix}
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:135]: Let's keep this macro definition with the rest of them at the top of the spec. Ordering may be important, so I'd make sure to put it after `_cc_name_suffix` is defined.

File: SPECS/openmpi/openmpi.spec [hunk 10]
```diff
@@ -0,0 +1,1139 @@
...
+%package -n python%{python3_pkgversion}-openmpi
+Summary:        OpenMPI support for Python 3
+Requires:       %{name} = %{version}-%{release}
+Requires:       python(abi)
+
+%description -n python%{python3_pkgversion}-openmpi
+OpenMPI support for Python 3.
+
+%prep
+%autosetup -p1 -n %{name}-%{version}
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:146]: Since we decided to keep `_cc_name_suffix`, which may modify the `Name` tag, this is probably going to be safer:
```suggestion
%autosetup -p1 -n openmpi-%{version}
```

See: comment for `Source0`.

File: SPECS/openmpi/openmpi.spec [hunk 11]
```diff
@@ -0,0 +1,1139 @@
...
+%ifarch %{java_arches}
+%bcond_without java
+%else
+%bcond_with java
+%endif
+# Private openmpi libraries
+%global __provides_exclude_from %{_libdir}/openmpi/lib/(lib(mca|ompi|open-(pal|rte|trace))|openmpi/).*.so
+%global __requires_exclude lib(mca|ompi|open-(pal|rte|trace)|vt).*
+# Run autogen - needed for some patches
+%bcond_with autogen
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:29]: Since it's unconditionally disabled, let's just remove the macro and the spec sections, which are now disabled:
```suggestion
```

File: SPECS/openmpi/openmpi.spec [hunk 12]
```diff
@@ -0,0 +1,1139 @@
...
+	--includedir=%{_includedir}/%{namearch} \
+	--sysconfdir=%{_sysconfdir}/%{namearch} \
+	--disable-silent-rules \
+	--enable-builtin-atomics \
+	--enable-mpi-cxx \
+	--enable-ipv6 \
+%if %{with java}
+	--enable-mpi-java \
+	--with-jdk-bindir=%{_libdir}/jvm/msopenjdk-17/bin \
+	--with-jdk-headers=%{_libdir}/jvm/msopenjdk-17/include \
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:165]: I don't see any `msopenjdk-17` in our code, so I'd be careful here. It looks like `msopenjdk-11` should be safe.

File: SPECS/openmpi/openmpi.spec [hunk 13]
```diff
@@ -0,0 +1,1139 @@
...
+%{_libdir}/%{name}/lib/openmpi/*
+%{_datadir}/modulefiles/mpi/
+%dir %{_libdir}/%{name}/share
+%dir %{_libdir}/%{name}/share/openmpi
+%{_libdir}/%{name}/share/openmpi/amca-param-sets
+%{_libdir}/%{name}/share/openmpi/help*.txt
+%if %{with rdma}
+%{_libdir}/%{name}/share/openmpi/mca-btl-openib-device-params.ini
+%endif
+%if 0%{?el7}
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:277]: Let's remove all sections for `e17` - I believe it's a distro-specific macro, which will never be true for Mariner.

File: SPECS/openmpi/openmpi.spec [hunk 14]
```diff
@@ -0,0 +1,1139 @@
...
+%{_libdir}/%{name}/lib/mpi.jar
+
+%files java-devel
+%{_libdir}/%{name}/bin/mpijavac
+%{_libdir}/%{name}/bin/mpijavac.pl
+# Currently this only contaings openmpi/javadoc
+%{_libdir}/%{name}/share/doc/
+%{_mandir}/%{namearch}/man1/mpijavac.1.gz
+%endif
+
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:323]:
```suggestion
```

File: SPECS/opensm/opensm.spec [hunk 15]
```diff
@@ -1,34 +1,39 @@
...
+Requires:       %{name}-libs%{?_isa} = %{version}-%{release}
+Requires:       logrotate
+Requires:       rdma
 Requires(post): systemd
-Requires(preun): systemd
 Requires(postun): systemd
+Requires(preun): systemd
 # RDMA is not currently built on 32-bit ARM: #1484155
-ExcludeArch: s390 %{arm}
+ExcludeArch:    s390 %{arm}
```
Comment (PawelWMS) [inline SPECS/opensm/opensm.spec:31]: We're not going to be building for these architectures, we might as well clean up and remove these lines.

File: SPECS/opensm/opensm.spec [hunk 16]
```diff
@@ -1,34 +1,39 @@
...
+%if %{__remake_config}
+BuildRequires:  autoconf
+BuildRequires:  automake
+BuildRequires:  libtool
+%endif
```
Comment (PawelWMS) [inline SPECS/opensm/opensm.spec:36]: Please move these BRs next to the other ones, so they are all in one place.

File: SPECS/orangefs/orangefs.spec [hunk 17]
```diff
@@ -1,10 +1,20 @@
-Vendor:         Microsoft Corporation
-Distribution:   Mariner
-Name: orangefs
-Version: 2.9.8
-Release: 1%{?dist}
-Summary: Parallel network file system client
-URL: https://www.orangefs.org/
+%global _hardened_build 1
```
Comment (PawelWMS) [inline SPECS/orangefs/orangefs.spec:1]: Not needed - we build this way by default.
```suggestion
%global _hardened_build 1
```
```suggestion
```

File: SPECS/orangefs/orangefs.spec [hunk 18]
```diff
@@ -1,10 +1,20 @@
...
+systems.  It provides very high performance access to disk storage for \
+parallel applications.  It is accessible through a variety of \
+interfaces, including the native OrangeFS library, the kernel, FUSE, \
+and MPI-IO. \
+\
+This package provides the pvfs2-client-core which is required to use \
+the kernel module.
+# Workaround for -fcommon issue
+# https://github.com/waltligon/orangefs/issues/80
+%define _legacy_common_support 1
```
Comment (PawelWMS) [inline SPECS/orangefs/orangefs.spec:13]: Seems unused in the spec. Let's try removing it and checking, if the package still builds.
```suggestion
```

File: SPECS/orangefs/orangefs.spec [hunk 19]
```diff
@@ -18,59 +28,51 @@ URL: https://www.orangefs.org/
...
-Source0: https://s3.amazonaws.com/download.orangefs.org/current/source/orangefs-%version.tar.gz
-Source1: orangefs-server.service
-Source2: orangefs-client.service
-Source3: orangefs.conf
-Source4: pvfs2tab
+License:        ASL 2.0 AND BSD AND GPLv2 AND LGPLv2+ AND LGPLv2 AND MIT AND OpenLDAP AND Public Domain AND zlib
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://www.orangefs.org/
+Source0:        https://s3.amazonaws.com/download.orangefs.org/current/source/orangefs-%{version}.tar.gz
```
Comment (PawelWMS) [inline SPECS/orangefs/orangefs.spec:35]:
```suggestion
Source0:        https://s3.amazonaws.com/download.orangefs.org/current/source/%{name}-%{version}.tar.gz
```

File: SPECS/orangefs/orangefs.spec [hunk 20]
```diff
@@ -18,59 +28,51 @@ URL: https://www.orangefs.org/
...
+BuildRequires:  make
+BuildRequires:  openssl-devel
+BuildRequires:  systemd
+BuildRequires:  perl(FindBin)
+BuildRequires:  perl(Getopt::Long)
+BuildRequires:  perl(Math::BigInt)
+BuildRequires:  perl(Term::ReadLine)
+%ifnarch armv7hl
+BuildRequires:  libibverbs-devel
+%endif
```
Comment (PawelWMS) [inline SPECS/orangefs/orangefs.spec:69]:
```suggestion
BuildRequires:  libibverbs-devel
```

File: SPECS/orangefs/orangefs.spec [hunk 21]
```diff
@@ -18,59 +28,51 @@ URL: https://www.orangefs.org/
...
 
 %description
-%desc
+%{desc}
 
 %prep
-%autosetup -p1 -n orangefs-v.%version
+%autosetup -p1 -n orangefs-v.%{version}
```
Comment (PawelWMS) [inline SPECS/orangefs/orangefs.spec:75]:
```suggestion
%autosetup -p1 -n %{name}-v.%{version}
```

File: SPECS/orangefs/orangefs.spec [hunk 22]
```diff
@@ -89,7 +91,7 @@ export LDFLAGS="%{optflags} -Wl,--as-needed"
     --enable-fuse --disable-usrint --with-db-backend=lmdb
 %else
 %configure --enable-external-lmdb --enable-shared --disable-static \
-   --enable-fuse --disable-usrint --with-db-backend=lmdb --with-openib=/usr
+   --enable-fuse --disable-usrint --with-db-backend=lmdb --with-openib=%{_prefix}
```
Comment (PawelWMS) [inline SPECS/orangefs/orangefs.spec:94]: Let's remove the part for `armv7hl`. And could you use only one `%configure` argument per line?

File: SPECS/orangefs/orangefs.spec [hunk 23]
```diff
@@ -188,13 +190,11 @@ install -p -m 644 %{SOURCE4} %{buildroot}%{_sysconfdir}
...
-
-Summary: Parallel network file system development libraries
-
-Requires: %{name}%{?_isa} = %{version}-%{release}
+Summary:        Parallel network file system development libraries
+Requires:       %{name}%{?_isa} = %{version}-%{release}
 
 %description devel
-%desc
+%{desc}
```
Comment (PawelWMS) [inline SPECS/orangefs/orangefs.spec:197]: Please move all `%package` and their `%description` sections between the default `%description` and the `%prep` section. That's the way we're trying to stick to.

File: SPECS/pmix/pmix.spec [hunk 24]
```diff
@@ -111,29 +96,25 @@ find %{buildroot} -name '*.la' | xargs rm -f
...
-%{_libdir}/libpmi2.so
+%{_libdir}/pkgconfig/*.pc
 
 %files tools
 %{_bindir}/*
 
 %changelog
+* Thu Feb 02 2023 Riken Maharjan <rmaharjan@microsoft.com> - 4.1.2-1
+- Move from Extended to core
+- Update to 4.1.2
```
Comment (PawelWMS) [inline SPECS/pmix/pmix.spec:115]: If we've used Fedora's spec as guidance for the upgrade, let's mention it here as well.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
A few more clean-up comments but looks good. Especially considering the amount of changes required to get `openmpi` to work!

File: SPECS/hwloc/hwloc.spec [hunk 25]
```diff
@@ -1,66 +1,66 @@
-Summary:   Portable Hardware Locality - portable abstraction of hierarchical architectures
-Name:      hwloc
-Version:   2.0.4
-Release:   5%{?dist}
-License:   BSD
+Summary:        Portable Hardware Locality - portable abstraction of hierarchical architectures
+Name:           hwloc
+Version:        2.5.0
```
Comment (PawelWMS) [inline SPECS/hwloc/hwloc.spec:3]: There seems to be one test failure for this package. We currently want to have all packages added to core pass their tests, so please take a look at the failure.

File: SPECS/munge/munge.spec (hunk 4 above)
Comment (rikenm1) [reply SPECS/munge/munge.spec]: updated.

File: SPECS/openmpi/openmpi.spec (hunk 9 above)
Comment (rikenm1) [reply SPECS/openmpi/openmpi.spec]: updated

File: SPECS/openmpi/openmpi.spec (hunk 12 above)
Comment (rikenm1) [reply SPECS/openmpi/openmpi.spec]: updated. 
I might have mixed msopenjdk and openjdk versions here.

File: SPECS/openmpi/openmpi.spec (hunk 13 above)
Comment (rikenm1) [reply SPECS/openmpi/openmpi.spec]: updated for openmpi.spec 
Looks like they were using internal pmix instead of external for el7. fixed.

File: SPECS/environment-modules/environment-modules.signatures.json (hunk 1 above)
Comment (rikenm1) [reply SPECS/environment-modules/environment-modules.signatures.json]: updated

File: SPECS/opensm/opensm.spec (hunk 15 above)
Comment (rikenm1) [reply SPECS/opensm/opensm.spec]: removed.

File: SPECS/opensm/opensm.spec (hunk 16 above)
Comment (rikenm1) [reply SPECS/opensm/opensm.spec]: updated.

File: SPECS/orangefs/orangefs.spec (hunk 22 above)
Comment (rikenm1) [reply SPECS/orangefs/orangefs.spec]: removed armv7hl and used only one line per configure.

File: SPECS/orangefs/orangefs.spec (hunk 23 above)
Comment (rikenm1) [reply SPECS/orangefs/orangefs.spec]: updated

File: SPECS/pmix/pmix.spec (hunk 24 above)
Comment (rikenm1) [reply SPECS/pmix/pmix.spec]: updated.

File: SPECS/hwloc/hwloc.spec (hunk 25 above)
Comment (rikenm1) [reply SPECS/hwloc/hwloc.spec:3]: @PawelWMS, this same test is also failing in the fedora. I just build their latest rawhide hwloc. I think this is coming from the upstream. 
I did search for the upstream patch but could not find anything. 
Might just have to let it sit like this for now.  
image
This is the same hwloc %check test that I ran on the fedora vm.

Comment (PawelWMS) [reply SPECS/hwloc/hwloc.spec:3]: @rikenm1, I think it actually succeeds in Fedora. At least it succeeded in their [latest build log for this package](https://kojipkgs.fedoraproject.org//packages/hwloc/2.5.0/5.fc38/data/logs/i686/build.log) (Fedora 38). I'd try digging deeper what might be causing the test to fail (check the logs or maybe there's a way to run the tests with more verbose output). Once we understand the failure, we can either correct it, or skip the test, if it's going to turn out, that it's not an issue on Mariner's end.

File: SPECS/openmpi/openmpi.spec [hunk 26]
```diff
@@ -143,10 +129,9 @@ Requires:       python(abi)
 OpenMPI support for Python 3.
 
 %prep
-%autosetup -p1 -n %{name}-%{version}
-%if %{with autogen}
+%autosetup -p1 -n openmpi-%{version}
 ./autogen.pl --force
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:133]: Since the original was building without `autogen`, then we should have removed this line along with `%if %{with autogen}`:
```suggestion
```

File: SPECS/pmix/pmix.spec [hunk 27]
```diff
@@ -112,7 +112,7 @@ find %{buildroot} -type f -name "*.la" -delete -print
 %changelog
 * Thu Feb 02 2023 Riken Maharjan <rmaharjan@microsoft.com> - 4.1.2-1
 - Move from Extended to core
-- Update to 4.1.2
+- Update to 4.1.2(from Fedora 38(license: MIT))
```
Comment (PawelWMS) [inline SPECS/pmix/pmix.spec:115]:
```suggestion
- Update to 4.1.2 (from Fedora 38 (license: MIT))
```

File: SPECS/orangefs/orangefs.spec [hunk 28]
```diff
@@ -86,13 +109,7 @@ autoreconf -vif -I maint/config
...
 %build
 export LDFLAGS="%{optflags} -Wl,--as-needed"
-%ifarch armv7hl
-%configure --enable-external-lmdb --enable-shared --disable-static \
-    --enable-fuse --disable-usrint --with-db-backend=lmdb
-%else
-%configure --enable-external-lmdb --enable-shared --disable-static \
-   --enable-fuse --disable-usrint --with-db-backend=lmdb --with-openib=%{_prefix}
-%endif
+%configure --enable-external-lmdb --enable-shared --disable-static --enable-fuse --disable-usrint --with-db-backend=lmdb --with-openib=%{_prefix}
```
Comment (PawelWMS) [inline SPECS/orangefs/orangefs.spec:112]: Thanks for the update. Could you also refactor to use one switch per line? This way it's easier to track config changes across commits. It's also going to make reading the config easier for everyone.:)

Comment (rikenm1) [reply SPECS/orangefs/orangefs.spec]: updated

File: SPECS/hwloc/hwloc.spec (hunk 25 above)
Comment (rikenm1) [reply SPECS/hwloc/hwloc.spec:3]: this failure is only happening on the docker side. docker by default blocks some of the sys_calls. the syscall that is failing here is move_page()

File: SPECS/hwloc/hwloc.spec [hunk 29]
```diff
@@ -70,6 +70,12 @@ Requires:       %{name}-plugins%{?_isa} = %{version}-%{release}
 
 %prep
 %autosetup -p1
+%if %{with_check}
+#Remove the failing test 
+sed -i '/hwloc_get_area_memlocation/d' tests/hwloc/Makefile.am
+#updating Makefile.am needs a rerun of autoreconf
+autoreconf 
+%endif
```
Comment (PawelWMS) [inline SPECS/hwloc/hwloc.spec:78]: I'd even run this always, so that we are sure to have the same build with and without the tests.
```suggestion
# Remove the failing test 
sed -i '/hwloc_get_area_memlocation/d' tests/hwloc/Makefile.am

# Updating Makefile.am needs a rerun of autoreconf
autoreconf 
```

File: SPECS/hwloc/hwloc.spec [hunk 30]
```diff
@@ -70,6 +70,12 @@ Requires:       %{name}-plugins%{?_isa} = %{version}-%{release}
 
 %prep
 %autosetup -p1
+%if %{with_check}
+#Remove the failing test 
+sed -i '/hwloc_get_area_memlocation/d' tests/hwloc/Makefile.am
```
Comment (PawelWMS) [inline SPECS/hwloc/hwloc.spec:75]: Please also mention why we're disabling this test in a comment.

File: SPECS/environment-modules/environment-modules.spec [hunk 31]
```diff
@@ -87,108 +90,91 @@ touch %{buildroot}%{_sysconfdir}/profile.d/modules.{csh,sh}
...
+rm -f %{buildroot}%{_docdir}/%{name}/{COPYING.GPLv2,ChangeLog-compat,INSTALL.txt,NEWS-compat}
 
+# install the rpm config file
 install -Dpm 644 contrib/rpm/macros.%{name} %{buildroot}/%{macrosdir}/macros.%{name}
 
 rm -rf %{buildroot}%{_datadir}/Modules/share/vim
 
 %check
-make test
+make test QUICKTEST=1
```
Comment (PawelWMS) [inline SPECS/environment-modules/environment-modules.spec:112]: Curious: why did you add `QUICKTEST=1`? What does this change?

File: SPECS/environment-modules/environment-modules.spec (hunk 2 above)
Comment (PawelWMS) [inline SPECS/environment-modules/environment-modules.spec:174]: I missed one more.;)

```suggestion
- Update to 5.2.0 (from Fedora 38 (license: MIT)).
```

File: SPECS/openmpi/openmpi.spec [hunk 32]
```diff
@@ -0,0 +1,1117 @@
...
+Requires:       %{name} = %{version}-%{release}
+Requires:       python(abi)
+
+%description -n python%{python3_pkgversion}-openmpi
+OpenMPI support for Python 3.
+
+%prep
+%autosetup -p1 -n openmpi-%{version}
+
+
```
Comment (PawelWMS) [inline SPECS/openmpi/openmpi.spec:134]: Nit-pick:
```suggestion
```

File: SPECS/hwloc/hwloc.spec (hunk 30 above)
Comment (rikenm1) [reply SPECS/hwloc/hwloc.spec:75]: updated

**Review by PawelWMS — APPROVED** [review]
